zict/tests/utils_test.py

```diff
@@ -131,3 +151,39 @@ def check_mapping(z):
 
 def check_closing(z):
     z.close()
+
+
+def check_bad_key_types(z, has_del=True):
+    """z does not accept any Hashable as keys.
+    Test that it reacts correctly when confronted with an invalid key type.
+    """
+    bad = object()
+
+    assert bad not in z
+    assert bad not in z.keys()
+    assert (bad, b"123") not in z.items()
+
+    with pytest.raises(TypeError):
+        z[bad] = b"123"
+    with pytest.raises(TypeError):
+        z.update({bad: b"123"})
+    with pytest.raises(KeyError):
+        z[bad]
+    if has_del:
+        with pytest.raises(KeyError):
+            del z[bad]
+
+
+def check_bad_value_types(z):
+    """z does not accept any Python object as values.
+    Test that it reacts correctly when confronted with an invalid value type.
+    """
+    bad = object()
```

Comment:
Why can't value be a Python `object`?

Reply:
Such is specifically the signature of LMDB, Zip, and File. They're not `MutableMapping[KT, VT]`; they're `MutableMapping[str, bytes]`.